src/function/utility/count_if.cpp

```diff
@@ -0,0 +1,79 @@
+#include "function/scalar_function.h"
+#include "function/utility/scalar_utility_functions.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace function {
+
+struct CountIf {
+    template<class T>
+    static inline void operation(T& input, int128_t& result) {
+        if (input != 0) {
+            result = 1;
+        } else {
+            result = 0;
+        }
+    }
+};
+
+static void getExecFunc(LogicalTypeID operandTypeID, scalar_func_exec_t& func) {
+    switch (operandTypeID) {
```

Comment:
I would reuse the Typeutils::visit() function instead of switch/case statements.